pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/ApexUnitTestClassShouldHaveAssertsRule.java

```diff
@@ -93,37 +93,45 @@ private Object checkForAssertStatements(ApexNode<?> node, Object data) {
             statements.addAll(blockStatement.findDescendantsOfType(ASTStatement.class));
             methodCalls.addAll(blockStatement.findDescendantsOfType(ASTMethodCallExpression.class));
         }
-        boolean isAssertFound = false;
-
-        for (final ASTMethodCallExpression methodCallExpression : methodCalls) {
-            if (ASSERT_METHODS.contains(methodCallExpression.getFullMethodName().toLowerCase(Locale.ROOT))) {
-                isAssertFound = true;
-                break;
-            }
-        }
-
-        // If we didn't find assert method invocations the simple way and we have a
-        // configured pattern, try it
+        boolean isAssertFound = isAssertMethodCalled(methodCalls);
+    
         if (!isAssertFound) {
             final String additionalAssertMethodPattern = getProperty(ADDITIONAL_ASSERT_METHOD_PATTERN_DESCRIPTOR);
-            final Pattern compiledPattern = getCompiledAdditionalAssertMethodPattern(additionalAssertMethodPattern);
-            if (compiledPattern != null) {
-                for (final ASTMethodCallExpression methodCallExpression : methodCalls) {
-                    final String fullMethodName = methodCallExpression.getFullMethodName();
-                    if (compiledPattern.matcher(fullMethodName).matches()) {
-                        isAssertFound = true;
-                        break;
-                    }
-                }
-            }
+            isAssertFound = isAssertMethodPatternMatched(methodCalls, additionalAssertMethodPattern);
         }
-
+    
         if (!isAssertFound) {
             addViolation(data, node);
         }
-
+    
         return data;
     }
+    
+    private boolean isAssertMethodCalled(List<ASTMethodCallExpression> methodCalls) {
+        for (final ASTMethodCallExpression methodCallExpression : methodCalls) {
+            if (ASSERT_METHODS.contains(methodCallExpression.getFullMethodName().toLowerCase(Locale.ROOT))) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean isAssertMethodPatternMatched(List<ASTMethodCallExpression> methodCalls, String pattern) {
```

Comment:
Change this method to:
`    private boolean isAssertMethodPatternMatched(List<ASTMethodCallExpression> methodCalls) {
        final String additionalAssertMethodPattern = getProperty(ADDITIONAL_ASSERT_METHOD_PATTERN_DESCRIPTOR);
        if (StringUtils.isNotBlank(additionalAssertMethodPattern)) {
            final Pattern compiledPattern = getCompiledAdditionalAssertMethodPattern(pattern);
            if (compiledPattern != null) {
                for (final ASTMethodCallExpression methodCallExpression : methodCalls) {
                    final String fullMethodName = methodCallExpression.getFullMethodName();
                    if (compiledPattern.matcher(fullMethodName).matches()) {
                        return true;
                    }
                }
            }
        }
        return false;
    }`